src/EFCore.PG/Query/ExpressionTranslators/Internal/NpgsqlDateTimeMemberTranslator.cs

```diff
@@ -44,7 +51,8 @@ public NpgsqlDateTimeMemberTranslator(NpgsqlSqlExpressionFactory sqlExpressionFa
                     new SqlExpression[] { _sqlExpressionFactory.Constant("day"), Now() },
                     nullable: true,
                     argumentsPropagateNullability: TrueArrays[2],
-                    returnType),
+                    returnType,
+                    _typeMappingSource.FindMapping(typeof(DateTime))),
```

Comment:
As below, date_trunc should just bubble up the mapping (which in this case would be `timestamp with time zone`).